bin/run_tests.sh

```diff
@@ -138,7 +138,14 @@ if [[ $code -ne 0 ]]; then
 fi
 if [[ "$CODECOV_TOKEN" != "" ]] && [[ "$CI" == "true" ]]; then
   echo "uploading all coverage data to codecov"
-  bash <(curl -s https://codecov.io/bash) -t $CODECOV_TOKEN
+  # code below adapted from https://docs.codecov.io/docs/about-the-codecov-bash-uploader
+  curl -s https://codecov.io/bash > codecov
+  CODECOV_VERSION=$(grep 'VERSION=\".*\"' codecov | cut -d'"' -f2);
+  for i in 1 256 512
+  do
+    diff <(shasum -a $i codecov) <(curl -s https://raw.githubusercontent.com/codecov/codecov-bash/$CODECOV_VERSION/SHA${i}SUM)  || { echo "CodeCov checksum validation failed!" >&2; exit 1; }
+  done
+  bash <codecov -t $CODECOV_TOKEN
```

Comment:
Denis centralized your script; let's use it from there.

```suggestion
  bash <(curl -s https://raw.githubusercontent.com/broadinstitute/codecov-bash-uploader/main/codecov-validated.bash)
```

Reply:
Sorry, just renamed it to `codecov-verified.bash` 😄